core/src/test/scala/org/apache/spark/deploy/IvyTestUtils.scala

```diff
@@ -243,16 +243,18 @@ private[deploy] object IvyTestUtils {
       withManifest: Option[Manifest] = None): File = {
     val jarFile = new File(dir, artifactName(artifact, useIvyLayout))
     val jarFileStream = new FileOutputStream(jarFile)
-    val manifest = withManifest.getOrElse {
-      val mani = new Manifest()
+    val manifest: Manifest = withManifest.getOrElse {
       if (withR) {
+        val mani = new Manifest()
         val attr = mani.getMainAttributes
         attr.put(Name.MANIFEST_VERSION, "1.0")
         attr.put(new Name("Spark-HasRPackage"), "true")
-      }
-      mani
+        mani
+      } else null
```

Comment:
I think typically Spark style is to have the else statement in separate line enclosed with `{}` like
```
} else {
   null
}
```